Support settings DJANGO_SETTINGS_MODULE in pytest_configure

See comment in #119, this should be possible:

```python
import os

def pytest_configure():
    os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'myproject.settings')
```

I still get the same `INTERNALERROR` traceback:
```python
INTERNALERROR> AssertionError: assert <django.conf.LazySettings object at 0x10eb0e4e0>.configured is True
```

I added a test in #192. If you import `django.setup` and call it after setting `DJANGO_SETTINGS_MODULE`, the test passes.

--- a/pytest_django/lazy_django.py
+++ b/pytest_django/lazy_django.py
@@ -22,8 +22,6 @@ def django_settings_is_configured():
 
     # If DJANGO_SETTINGS_MODULE is defined at this point, Django is assumed to
     # always be loaded.
-    from django.conf import settings
-    assert settings.configured is True
     return True
 
 

--- a/tests/test_django_settings_module.py
+++ b/tests/test_django_settings_module.py
@@ -96,6 +96,25 @@ def test_ds_after_user_conftest(testdir, monkeypatch):
     result.stdout.fnmatch_lines(['*1 passed*'])
 
 
+def test_ds_in_pytest_configure(testdir, monkeypatch):
+    monkeypatch.delenv('DJANGO_SETTINGS_MODULE')
+    pkg = testdir.mkpydir('tpkg')
+    settings = pkg.join('settings_ds.py')
+    settings.write(BARE_SETTINGS)
+    testdir.makeconftest("""
+        import os
+
+        from django.conf import settings
+
+        def pytest_configure():
+            if not settings.configured:
+                os.environ.setdefault('DJANGO_SETTINGS_MODULE',
+                                      'tpkg.settings_ds')
+    """)
+    r = testdir.runpytest()
+    assert r.ret == 0
+
+
 def test_django_settings_configure(testdir, monkeypatch):
     """
     Make sure Django can be configured without setting

